choreolib/py/choreolib/choreoTrajectory.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+import math
+
+from wpimath.geometry import Pose2d, Rotation2d
+from wpimath.kinematics import ChassisSpeeds
+
+# Doesn't appear to be pulled into python from Interpolatable
+def _floatInterp(start, end, frac) -> float:
+    return start + (end - start) * frac
+
+
+class ChoreoTrajectoryState:
+    def __init__(
+        self,
+        timestamp: float,
+        x: float,
+        y: float,
+        heading: float,
+        velocityX: float,
+        velocityY: float,
+        angularVelocity: float,
+    ):
+        self.timestamp = timestamp
+        self.x = x
+        self.y = y
+        self.heading = heading
+        self.velocityX = velocityX
```

Comment:
These names should ideally be snake case to follow https://pep8.org/#naming-conventions, though robotpy doesn't follow that standard.

Reply:
robotpy mostly matches java for wrapped stuff /shrug

Reply:
ctre has chosen to do snake case for their python stuff so I guess it's whatever now